ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -450,6 +459,98 @@ func checkResponse(path string, code int, body []byte) error {
 	return fmt.Errorf("IPFS-post '%s' unsuccessful: %d: %s", path, code, body)
 }
 
+// postCtxStreaming makes a POST request against
+// the ipfs daemon, reads the body of the response line by line
+// at evey millisecond. It cancels the context if timeout duration
+// has passed since last new data on the buffer.
+func (ipfs *Connector) postCtxStreaming(ctx context.Context, path string, timeout time.Duration) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	elapsed := time.Now()
+
+	go func() {
+		for {
+			select {
+			case <-timer.C:
+				{
+					if time.Since(elapsed) >= timeout {
+						cancel()
```

Comment:
I could have explicitly returned once context is done, but I have relied on reader to end by throwing `context cancelled` error